src/ray/gcs/tables.cc

```diff
@@ -339,6 +339,157 @@ std::string Set<ID, Data>::DebugString() const {
   return result.str();
 }
 
+template <typename ID, typename Data>
+Status Hash<ID, Data>::Update(const DriverID &driver_id, const ID &id,
+                              const DataMap &data_map, const HashCallback &done) {
+  num_adds_++;
+  auto callback = [this, id, data_map, done](const CallbackReply &reply) {
+    if (done != nullptr) {
+      (done)(client_, id, data_map);
+    }
+  };
+  flatbuffers::FlatBufferBuilder fbb;
+  std::vector<flatbuffers::Offset<flatbuffers::String>> data_vec;
+  data_vec.reserve(data_map.size() * 2);
+  for (auto const &pair : data_map) {
+    // Add the key.
+    data_vec.push_back(fbb.CreateString(pair.first));
+    flatbuffers::FlatBufferBuilder fbb_data;
+    fbb_data.ForceDefaults(true);
+    fbb_data.Finish(Data::Pack(fbb_data, pair.second.get()));
+    std::string data(reinterpret_cast<char *>(fbb_data.GetBufferPointer()),
+                     fbb_data.GetSize());
+    // Add the value.
+    data_vec.push_back(fbb.CreateString(data));
+  }
+
+  fbb.Finish(CreateGcsTableEntry(fbb, GcsTableNotificationMode::APPEND_OR_ADD,
```

Comment:
I'm not sure if it makes sense to overload `GcsTableEntry` with hash table semantics. What do you think about creating a separate flatbuffer type specific for hash table operations?

Reply:
Maybe the name `GcsTableEntry` should be renamed to `GcsEntry` to make it more meaningful to hash. At beginning, I create a `GcsHashEntry` structure. After I checked [`HGETALL document`](https://redis.io/commands/hgetall) which returns 2*n list. I decided to reuse `GcsEntry` since it better aligns to the redis semantic.